app/src/main/java/ch/epfl/sweng/zuluzulu/Fragments/ChannelFragment.java

```diff
@@ -71,7 +75,7 @@ public static ChannelFragment newInstance(User user) {
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         if (getArguments() != null) {
-            user = (User) getArguments().getSerializable(ARG_USER);
+            user = (AuthenticatedUser) getArguments().getSerializable(ARG_USER);
```

Comment:
You can see at my pull request #95 
You will soon be able to check if to check the user role (user, moderator, admin...) !
Also you should maybe check if the user is connected before casting to authenticatedUser. In case there is a bug in the menu...

Reply:
I agree with you, I let you merge before hence I will be able to improve my code.
xDD

Reply:
Ok !

I didn'it implement the verification in any fragment yet, but it should work !

May be this code should work in a fragment :
```
if(!user.hasRole(UserRole.USER)) {
    getActivity().getSupportFragmentManager().beginTransaction().remove(this).commit();
} else {
//... casting etc...
}
```